language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/BallerinaTextDocumentService.java

```diff
@@ -439,7 +443,18 @@ public CompletableFuture<List<? extends TextEdit>> formatting(DocumentFormatting
                 if (syntaxTree.isEmpty()) {
                     return Collections.emptyList();
                 }
-                String formattedSource = Formatter.format(syntaxTree.get()).toSourceCode();
+                String formattedSource;
+                if (context.currentModule().isPresent() &&
+                        !(context.currentModule().get().project().kind() == ProjectKind.SINGLE_FILE_PROJECT)) {
```

Comment:
Extract `context.currentModule()` out to a variable. IIRC, you'll get a sonar check failure for this too.